tests/ga/test_update.py

```diff
@@ -1597,6 +1600,15 @@ def test_it_should_recreate_handler_env_on_service_startup(self):
             self.assertIn(HandlerEnvironment.eventsFolder, content[0][HandlerEnvironment.handlerEnvironment],
                           "{0} not found in HandlerEnv file".format(HandlerEnvironment.eventsFolder))
 
+    def _add_write_permission_to_goal_state_files(self):
+        # UpdateHandler.run() marks some of the files from the goal state as read-only. Those files are overwritten when
+        # a new goal state is fetched. This is not a problem for the agent, since it  runs as root, but tests need
+        # to make those files writtable before fetching a new goal state. Note that UpdateHandler.run() fetches a new
+        # goal state, so tests that make multiple calls to that method need to call this function in-between calls.
+        for gb in READONLY_FILE_GLOBS:
+            for path in glob.iglob(os.path.join(conf.get_lib_dir(), gb)):
+                fileutil.chmod(path, stat.S_IRUSR | stat.S_IWUSR)
```

Comment:
Out of curiosity, if the tests are not running as root, how are we able to change their permissions? Can the same user change permissions for a read-only file if non-root?

Reply:
Yup, setting the read bit does not affect chmod, so all ok with the tests